rtos/Mutex.h

```diff
@@ -89,7 +106,8 @@ class Mutex : private mbed::NonCopyable<Mutex> {
 
       @note You cannot call this function from ISR context.
      */
-    osStatus lock(uint32_t millisec=osWaitForever);
+    MBED_DEPRECATED_SINCE("mbed-os-5.10.0", "Replaced with try_lock(millisec) function")
```

Comment:
`trylock()` and `trylock_for(millisec)`, and maybe mention `lock()` too. 

(That covers 0, finite and osWaitForever as the possible parameters).

Reply:
fixed.